testing/integration/broker/broker_suite_test.go

```diff
@@ -1,70 +1,98 @@
+
+	sqsClientConfig, err := sqs.NewConfig(config.Provider)
+	Expect(err).ToNot(HaveOccurred())
+
+	// by default the integration tests run without a permission boundary so
+	// that there are no dependencies on setting up external IAM policies
+	// to run the test with a predefined permission boundary policy set the following environment variable:
+	//
+	// PERMISSIONS_BOUNDARY_ARN="arn:aws:iam::ACCOUNT-ID:policy/SQSBrokerUserPermissionsBoundary"
+	//
+	optionalPermissionsBoundary := os.Getenv("PERMISSIONS_BOUNDARY_ARN")
+	if optionalPermissionsBoundary != "" {
+		sqsClientConfig.PermissionsBoundary = optionalPermissionsBoundary
 	}
-	return actualState == matcher.expectedState, nil
-}
 
-func (matcher *haveLastOperationStateMatcher) FailureMessage(actual interface{}) (message string) {
-	return fmt.Sprintf("Expected\n\t%#v\nto have last operation state of\n\t%#v", actual, matcher.expectedState)
-}
+	logger := lager.NewLogger("sqs-service-broker-test")
+	logger.RegisterSink(lager.NewWriterSink(GinkgoWriter, config.API.LagerLogLevel))
 
-func (matcher *haveLastOperationStateMatcher) NegatedFailureMessage(actual interface{}) (message string) {
-	return fmt.Sprintf("Expected\n\t%#v\nnot to have last operation state of\n\t%#v", actual, matcher.expectedState)
-}
+	sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(sqsClientConfig.AWSRegion)}))
 
-func (matcher *haveLastOperationStateMatcher) MatchMayChangeInTheFuture(actual interface{}) bool {
-	actualState, err := matcher.state(actual)
-	if err != nil {
-		return false
+	sqsProvider := &sqs.Provider{
+		Client: struct {
+			*secretsmanager.SecretsManager
+			*cloudformation.CloudFormation
+		}{
+			SecretsManager: secretsmanager.New(sess),
+			CloudFormation: cloudformation.New(sess),
+		},
+		Environment:         sqsClientConfig.DeployEnvironment,
+		ResourcePrefix:      sqsClientConfig.ResourcePrefix,
+		PermissionsBoundary: sqsClientConfig.PermissionsBoundary,
+		Timeout:             sqsClientConfig.Timeout,
+		Logger:              logger,
 	}
-	switch actualState {
-	case domain.InProgress:
-		return true
-	default:
-		return false
+
+	serviceBroker, err := brokerbase.New(config, sqsProvider, logger)
+	Expect(err).ToNot(HaveOccurred())
+	brokerAPI := brokerbase.NewAPI(serviceBroker, logger, config)
+
+	broker = brokertesting.New(brokerapi.BrokerCredentials{
+		Username: "username",
+		Password: "password",
+	}, brokerAPI)
+})
+
+// DescribeIntegrationTest acts like Describe but only conditionally runs tests
+// based on if the ENABLE_INTEGRATION_TESTS enviironment variable is set.  This
+// results in tests showing up as "pending" rather than as "passed" (which can
+// be misleading when viewing the test output)
+func DescribeIntegrationTest(desc string, fn func()) (ok bool) {
+	if os.Getenv("ENABLE_INTEGRATION_TESTS") == "true" {
+		ok = Describe(desc, fn)
+	} else {
+		ok = PDescribe(desc, fn)
 	}
+	return ok
+}
+
+func TestBroker(t *testing.T) {
```

Comment:
I like the idea of using `Short()` (I didn't know that was a thing!) :+1: 

however I do not want to go back to using "Skip()" ... the reason I moved away from that was because I got tripped up by thinking the integration tests were passing and wanted a way to run the tests so that they definitely fail ....  this is now possible using `--failOnPending` to ensure all the tests that should exist/run have a passing state

Reply:
Yeah makes sense, this sounds like a good compromise 👍

Reply:
Awww I am unable to using `testing.Short()` as I need due to the ginkgo stuff executing before the `init()` stuff in testing is executed. :cry: 

I'm gonna leave it as is for now, since it's probably not worth the effort, but I agree with the sentiment that there must be a nicer way than either using environment variables or forcing people to explicitly name the package to test every time this comes up

Reply:
The only other option I can suggest is build tags